plutus-core/plutus-ir/test/PlutusIR/Transform/RewriteRules/Tests.hs

```diff
@@ -0,0 +1,23 @@
+module PlutusIR.Transform.RewriteRules.Tests where
+
+import PlutusCore.Quote
+import PlutusIR.Parser
+import PlutusIR.Test
+import PlutusIR.Transform.RewriteRules as RewriteRules
+
+import Data.Default.Class
+import Test.Tasty
+import Test.Tasty.Extras
+
+test_RewriteRules :: TestTree
+test_RewriteRules = runTestNestedIn ["plutus-ir/test/PlutusIR/Transform"] $
+    testNested "RewriteRules" $
+    fmap
+        (goldenPir (runQuote . RewriteRules.userRewrite def)  pTerm)
+        [ "equalsInt" -- this tests that the function works on equalInteger
+        , "divideInt" -- this tests that the function excludes not commutative functions
+        , "multiplyInt" -- this tests that the function works on multiplyInteger
```

Comment:
I'd consider adding a property test that tests whether optimizing a term permutes all bulitin applications that are supposed to be permuted and leaves others intact, but this is quite a bit more work, so having several golden tests is probably enough indeed.